src/internal/signals/signals_test.go

```diff
@@ -0,0 +1,45 @@
+package signals_test
+
+import (
+	"os"
+	"os/signal"
+	"testing"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/signals"
+)
+
+func TestSignals(t *testing.T) {
+	if err := testSignal(os.Interrupt); err != nil {
+		t.Error(err)
+	}
+}
+
+func testSignal(s os.Signal) error {
+	var (
+		c = make(chan os.Signal, 1)
+		g errgroup.Group
+	)
+	signal.Notify(c, signals.TerminationSignals...)
+	defer signal.Stop(c)
+	defer close(c)
+
+	g.Go(func() error {
+		p, err := os.FindProcess(os.Getpid())
+		if err != nil {
+			return errors.Wrap(err, "could not find own process")
+		}
+		return errors.Wrap(p.Signal(s), "could not signal self")
+	})
+
+	g.Go(func() error {
```

Comment:
looks good. out of curiosity, is there any value in this being in a goroutine?

Reply:
I think that it helps to think about signalling in one goroutine and handling signals in another, and I do like the consolidated error handling with `errgroup.Group`, but it’s not technically required.